nexus/reconfigurator/planning/src/planner.rs

```diff
@@ -2056,6 +2061,43 @@ mod test {
             TEST_NAME,
         );
 
+        // Now remove the decommissioned sled from the input entirely. (This
+        // should not happen in practice at the moment -- entries in the sled
+        // table are kept forever -- but we need to test it.)
```

Comment:
To clarify, would you expect the `sleds_removed` to be non-zero later?

Reply:
So there's several steps to this:

1. Sled becomes decommissioned.
2. Zones that no longer have resources allocated to them are cleaned up from the blueprint. (This is what I'm implementing right now.)
3. Decommissioned sleds with no zones are cleaned up from the blueprint. (This would make `sleds_removed` non-zero.)
4. Decommissioned sleds that are no longer relevant are removed from the sled table.

So yes, `sleds_removed` will become non-zero with step 3, but that will come before the sled entries go away from the planning input.

I'll add a comment to this effect.